sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -324,7 +337,32 @@ static int sof_audio_probe(struct platform_device *pdev)
 
 	dmi_check_system(sof_rt5682_quirk_table);
 
-	if (sof_rt5682_quirk & SOF_RT5682_SSP1) {
+	dev_dbg(&pdev->dev, "sof_rt5682_quirk = %lx\n",
+		sof_rt5682_quirk);
+
+	if (sof_rt5682_quirk & SOF_RT5682_LEGACY_PLATFORM) {
+		/*
+		 * "SSP2-Codec" is used on bytcr machine driver
+		 * no matter which port is used
+		 */
+		snprintf(sof_rt5682_dai_link_name,
+			 sizeof(sof_rt5682_dai_link_name),
+			 "%s", "SSP2-Codec");
+		if (sof_rt5682_quirk & SOF_RT5682_SSP0) {
+			snprintf(sof_rt5682_cpu_dai_name,
+				 sizeof(sof_rt5682_cpu_dai_name),
+				 "%s", "ssp0-port");
+		} else {
+			/* set ssp2-port by default */
+			snprintf(sof_rt5682_cpu_dai_name,
+				 sizeof(sof_rt5682_cpu_dai_name),
+				 "%s", "ssp2-port");
+		}
+		sof_rt5682_dais[0].name = sof_rt5682_dai_link_name;
+		sof_rt5682_dais[0].cpu_dai_name = sof_rt5682_cpu_dai_name;
+		/* Only ssp port is supported on legacy platforms */
+		sof_audio_card_rt5682.num_links = 1;
```

Comment:
it's a bit hacky. Could we break the dai_links in two and add the hdmi parts in the else case (similar to how it's done with bytcr_rt5640).

Reply:
@plbossart Sorry, I didn't see it on bytcr_rt5640, but I think we can do it similar to how nocodec machine driver do.

Reply:
Sorry, yes I was confused. The bytcr_rt5640 adds the routes dynamically, not the links. My bad. what you've done is actually the same as in skl_hda_dsp_generic.c, so I guess it's ok. If you find something nicer then it's good, otherwise let's keep this.